packages/hub/src/app/models/[owner]/[slug]/SquiggleSnippetDraftDialog.tsx

```diff
@@ -0,0 +1,139 @@
+import { FC, useState } from "react";
+import { graphql, useFragment } from "react-relay";
+
+import { Button, Modal, TextTooltip } from "@quri/ui";
+
+import { SquiggleSnippetDraftDialog_Model$key } from "@/__generated__/SquiggleSnippetDraftDialog_Model.graphql";
+import { SquiggleSnippetFormShape } from "./EditSquiggleSnippetModel";
+import { useClientOnlyRender } from "@/hooks/useClientOnlyRender";
+
+export type Draft = {
+  formState: SquiggleSnippetFormShape;
+  version: string;
+};
+
+type DraftLocator = {
+  ownerSlug: string;
+  modelSlug: string;
+};
+
+function draftKey(key: DraftLocator) {
+  return `modelDraft:${key.ownerSlug}/${key.modelSlug}`;
+}
+
+function localStorageExists() {
+  return Boolean(typeof window !== "undefined" && window.localStorage);
+}
+
+export const draftUtils = {
+  exists(locator: DraftLocator): boolean {
+    if (!localStorageExists()) {
+      return false;
+    }
+    return Boolean(window.localStorage.getItem(draftKey(locator)));
+  },
+  save(locator: DraftLocator, draft: Draft) {
+    if (!localStorageExists()) {
+      return;
+    }
+    window.localStorage.setItem(draftKey(locator), JSON.stringify(draft));
+  },
+  load(locator: DraftLocator): Draft | null {
+    if (!localStorageExists()) {
+      throw new Error("Local storage not available");
+    }
+    const value = window.localStorage.getItem(draftKey(locator));
+    if (!value) return null;
+    return JSON.parse(value); // TODO - validate
```

Comment:
Just to confirm that the TODO is not meant for this PR

Reply:
Yep, I mentioned this in the PR description:
> I don't validate that the restored draft matches the form shape; security implications are not significant [...]

It still might be useful as a general practice, but doesn't matter.